### PR TITLE
feat(secret): guard against removing last key and add single-key update/delete e2e tests

### LINKED ISSUE #406: Support updating or deleting a single key within a secret
## Summary
Allow users to update or delete a single key/value pair within an existing secret without replacing the entire secret payload.

## Problem
Currently, modifying a secret requires replacing the full JSON object. This is error-prone and inconvenient when only a single environment variable within the secret needs to change — for example, rotating one credential or removing a deprecated key.

## Proposed Scope
- Add an `update` subcommand (or flag) that accepts a key and value and merges it into the existing secret payload.
- Add a `delete` subcommand (or flag) that accepts a key name and removes it from the existing secret payload.
- Both operations should read the current secret, apply the change, and write back the updated JSON object.
- Existing keys not referenced in the operation must be preserved unchanged.

## Proposed CLI Shape (example)

```sh
# Update a single key
env-secrets aws secret update MY_SECRET KEY=newvalue

# Delete a single key
env-secrets aws secret delete MY_SECRET KEY
```

The exact command shape should stay consistent with the existing provider UX.

## Acceptance Criteria
- A single key can be updated in an existing secret without affecting other keys.
- A single key can be deleted from an existing secret without affecting other keys.
- Attempting to update/delete a key in a non-existent secret produces a clear error.
- Attempting to delete a key that does not exist in the secret produces a clear error.
- Unit tests cover update and delete paths, including edge cases (missing secret, missing key, last remaining key).
- Documentation and help text are updated.

## Notes
- Deleting the last remaining key in a secret should either error with a clear message or prompt the user — deleting a secret entirely should remain an explicit separate action.
- This applies to the AWS provider initially; follow-on issues can extend to additional providers as they are added.

### PR DESCRIPTION
**Pull request by markcallen**
## Summary

Closes #406

- Extracts the key-removal loop from the \`remove\` command into a new \`applyKeyRemovals\` helper in \`src/cli/helpers.ts\` so it is unit-testable
- Adds a **last-key guard**: attempting to remove all keys from a JSON secret now fails with a clear error directing the user to \`aws secret delete\` instead
- Adds 5 unit tests for \`applyKeyRemovals\` covering happy path, partial-miss, none-found, and last-key scenarios
- Adds 5 e2e tests against LocalStack covering: append to non-existent secret, remove missing key, remove last key, append-overwrite preserving other keys, and partial remove with missing-key reporting
- Updates \`docs/AWS.md\` to document \`append\` and \`remove\` as the way to update/delete individual keys, with an explicit error-case reference table and the last-key guard note

## Test plan

- [ ] \`npx jest --testPathPattern="helpers"\` — all unit tests pass (31 total, 5 new)
- [ ] \`npm run test:e2e\` — new e2e tests pass against LocalStack (requires Docker)
- [ ] \`npm run build\` — TypeScript compiles cleanly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli/helpers.ts [hunk 1]
```diff
@@ -234,6 +234,38 @@ export const resolveOutputFormat = (
...
+
+  if (removed.length === 0) {
+    throw new Error(
+      `None of the requested keys exist in secret "${secretName}".`
+    );
+  }
+
+  if (Object.keys(payload).length === 0) {
+    throw new Error(
+      `Cannot remove all keys from secret "${secretName}" — it would leave an empty object. Delete the secret explicitly using 'aws secret delete'.`
```
[the remark this answers is not in the document]
Comment (markcallen) [reply src/cli/helpers.ts]: Fixed in 4b8b0cb — updated the error message to say `'env-secrets aws secret delete'` instead of `'aws secret delete'`.

File: docs/AWS.md [hunk 2]
```diff
@@ -241,16 +241,33 @@ source secrets.env
...
    ```
 
+   **Error cases:**
+
+   | Situation                                                      | Behaviour                                                                                            |
+   | -------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- |
+   | Secret does not exist                                          | Error from AWS (`ResourceNotFoundException`)                                                         |
+   | `remove --key` targets a key not in the secret                 | Error: `None of the requested keys exist in secret "…"`                                              |
+   | `remove` would empty the secret (last key)                     | Error: `Cannot remove all keys from secret "…"` — use `aws secret delete` to remove the whole secret |
+   | Partial match (`remove` with some valid and some missing keys) | Succeeds; output shows `RemovedKeys` and `MissingKeys`                                               |
```
[the remark this answers is not in the document]
Comment (markcallen) [reply docs/AWS.md]: Fixed in 4b8b0cb — updated the error cases table to say `env-secrets aws secret delete`.

File: docs/AWS.md [hunk 3]
```diff
@@ -299,6 +316,7 @@ source secrets.env
 - `create`, `update`, and `append` accept `--value`, `--value-stdin`, or `--file` (use only one).
 - `create` always stores `SecretString` as a JSON object.
 - `append` and `remove` require the secret value to be a JSON object.
+- `remove` will error if it would leave the secret with zero keys — use `aws secret delete` to remove the entire secret.
```
[the remark this answers is not in the document]
Comment (markcallen) [reply docs/AWS.md]: Fixed in 4b8b0cb — updated the safety note bullet to say `env-secrets aws secret delete`.
